### DISCUSSION TITLE
Add Option for Directory-Level File Access Permission

### DISCUSSION THREAD

**Discussion by DotNetTitan** [cline/cline#365] [Feature Requests]
Currently, when using Claude Dev in VS Code, the user is required to manually grant permission for the AI to read and modify each individual file. While this adds security, it can become tedious when working with larger projects involving multiple files.

**Feature Request:** I would like to request an enhancement to the permission system, allowing users to grant file access at the directory level instead of individual files. This feature could be implemented as an option to select a directory from which Claude Dev can read and modify all files within it. The user would grant blanket permission for this selected directory, streamlining workflows and reducing the need for repetitive permissions.

**Suggested Implementation**:

    Provide an option during the first permission request that allows the user to select a directory for Claude Dev to access.

This would be a major quality-of-life improvement for users who work on larger projects or prefer to organize files into specific directories.

Thank you for considering this enhancement!

**Comment by jakenuts** [comment]
I'd love this. It would also allow for saving tokens/effort by using search/replace bulk operations for changes across many files that could be done using cli tools or VS Code find/replace commands.
